### PR TITLE
Move fatty project index controller code to service

### PR DESCRIPTION
**Pull request by shalapatil**
## Notion card
https://www.notion.so/saeloun/Refactor-controllers-51ad6bf864c448308db7a9c6f65241ad

## Summary
View was doing DB query. Moved that to service and all required calls from controller to the service. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [ ] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

**Comment by shalapatil** [comment]
CI Test Suite was passing before merging latest develop,
Now there is no CI, buildkite is failing.

File: app/services/projects_index_service.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class ProjectsIndexService
```
Comment (vipulnsward) [inline app/services/projects_index_service.rb:3]: Index has a different meaning for us as we are using ES, so we should prefer using "Fetch" or similar

File: app/services/projects_index_service.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        params[:client_id], params[:user_id], params[:billable],
+        params[:search])
+    end
+
+    def set_clients
+      @clients = current_company.clients.kept
+    end
+
+    def set_users
+      @users = current_company.employments.joins(:user)
```
Comment (vipulnsward) [inline app/services/projects_index_service.rb:36]: Went over this, its fine for now

**Comment by shalapatil** [comment]
@apoorv1316 Please test this on staging.

**Pull request merged**